plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeBucketFunction.java

```diff
@@ -51,13 +51,14 @@ public DeltaLakeBucketFunction(TypeOperators typeOperators, List<DeltaLakeColumn
     @Override
     public int getBucket(Page page, int position)
     {
-        long result = 0;
+        long hash = 0;
         for (int channel = 0; channel < page.getChannelCount(); channel++) {
             Block block = page.getBlock(channel);
             Object value = TypeUtils.readNativeValue(types.get(channel), block, position);
-            result += hashValue(hashCodeInvokers.get(channel), value);
+            long valueHash = hashValue(hashCodeInvokers.get(channel), value);
+            hash = (31 * hash) + valueHash;
         }
-        return (int) ((result & Long.MAX_VALUE) % bucketCount);
+        return (int) ((hash & Long.MAX_VALUE) % bucketCount);
     }
 
     private static long hashValue(MethodHandle method, Object value)
```

Comment:
I wonder if NULL_HASH_CODE should be -1 or some other less frequent number than 0, as 0 is quite common.

Reply:
we have 4x NULL_HASH_CODE constant in the code.
i think we can change that separately.